src/main/resources/avro/readmethods.avdl

```diff
@@ -14,8 +14,10 @@ to `ReferenceSet`s containing that same `Reference`. If no reference is
 specified, all `ReadGroup`s must be aligned to the same `ReferenceSet`.
 */
 record SearchReadsRequest {
-  /** If nonempty, restrict this query to reads within the given readgroups. */
-  array<string> readGroupIds = [];
+  /**
+  The ReadGroup to search.
+  */
+  string readGroupId;
```

Comment:
Two high-level thoughts:
a) when designing an API from scratch, I agree that we should make thoughtful tradeoffs between extra work for client developers and extra work for server developers, since the overall goal of wide adoption needs both groups to be happy. I'm open to different answers in different circumstances. My general bias would be towards (1) -- when in doubt, keep the server simple -- since it's always possible to add more server options later, but it's hard to remove them. I'd tilt towards (2) -- tell the server enough to let it optimize for you -- when, as @jeromekelleher says, there's a "compelling performance argument".

b) we're not starting from scratch here -- we have a published API with existing server implementations and existing client apps. Making backwards-incompatible changes is a big deal -- even if they only break a few clients, those are by definition our most eager early adopters. If we set the expectation that coding to the GA4GH API is risky, we'll be discouraging people from ever getting on board.

If I read the PR right, there are about a dozen independent changes proposed here. I defer to CH and others on which of them are likely to break existing clients, vs. still being in the "from scratch" state.